design/security/high_availability.md

```diff
@@ -0,0 +1,66 @@
+# High availability
+
+The Obscuro Sequencer must have HA capabilities. The reasoning is covered in the "Fast finality" design docs.
+
+The requirement is that the service must continue even if the enclave of the sequencer crashes and is corrupted.
+
+One option is to require a governance event to whitelist a new sequencer enclave in case the current one becomes unavailable.
+This is not a good solution, because it would lead to significant downtime.
+
+The preferred option is to mark multiple enclaves as sequencers, and make it the responsibility of the operator to switch
+between them to ensure that service goes on uninterrupted.
+
+## MEV Protection 
+
+The Sequencer is the only Obscuro node capable of producing Light Batches, and thus the only node that can in theory
+attempt to extract value via MEV. If the operator has multiple enclaves available it can both provide an uninterrupted service
+and extract some value at the same time.
+
+Introducing startup delays does not help too much in this case, because the operator could hold key transactions for longer.
+
+
+An alternative solution is to introduce restart transparency, such that the Obscuro network can assess the likelyhood
+of bad behaviour.  
+
+On a high level, all enclave restart events of all the sequencer whitelisted enclaves must be published to the network.
+
+### Protocol
+
+Each Batch will have three elements:
+
+1. The header
+2. The transaction payload
+3. The protocol payload
+
+The protocol payload will not be included in the rollup published to the data availability layer. 
```

Comment:
So I guess the protocol payload is not included in the header hash? (To ensure header hash can be verified from rollup data).

Unlike (I think) eth log data I think this payload cannot be reproduced. But do we need a hash that includes it too for any reason?

Reply:
good point. need to clarify this